toMatch output is truncated

### Describe the bug

While using e.g. `toMatch` the expected result is truncated even when setting `outputTruncateLength` to 1000

```node
 FAIL  test/git.spec.ts > git > random seeded tests for seed 3694299162975756 > commitEntry > should return a valid random commitEntry
AssertionError: expected 'Author: Reynold71 <Reynold_Walsh57@ya…' to match /^Author: [\w_\. ]+ \<[\w\.]+@[\w\.]+\>$/
 ❯ test/git.spec.ts:62:30
     60|             expect(parts[5]).toMatch(/^\s{4}.+$/);
     61|           } else {
     62|             expect(parts[1]).toMatch(/^Author: [\w_\. ]+ \<[\w\.]+@[\w\.]+\>$/);
       |                              ^
     63|             expect(parts[2]).toMatch(/^Date: .+$/);
     64|             expect(parts[3]).toBe('');
```

It might be somewhat related to #2448
But there is no expected vs actual comparison/diff, so it's harder here to find the problem of the failing test

### Reproduction

will add if really needed

### System Info

```shell
System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 3.49 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.4.0 - ~/.nvm/versions/node/v18.4.0/bin/node
    npm: 8.12.1 - ~/.nvm/versions/node/v18.4.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Safari: 16.2
  npmPackages:
    @vitest/coverage-c8: ~0.28.3 => 0.28.3 
    @vitest/ui: ~0.28.3 => 0.28.3 
    vite: ~4.0.4 => 4.0.4 
    vitest: ~0.28.3 => 0.28.3
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/docs/api/expect.md
+++ b/docs/api/expect.md
@@ -498,9 +498,10 @@ type Awaitable<T> = T | PromiseLike<T>
   })
   ```
 
-::: tip NOTE
-You can set [chaiConfig](/config/#chaiconfig-truncatethreshold) configuration value to truncate the error assertion message.
+::: tip
+If the value in the error message is too truncated, you can increase [chaiConfig.truncateThreshold](/config/#chaiconfig-truncatethreshold) in your config file.
 :::
+
 ## toMatchObject
 
 - **Type:** `(received: object | array) => Awaitable<void>`

--- a/docs/api/expect.md
+++ b/docs/api/expect.md
@@ -498,6 +498,9 @@ type Awaitable<T> = T | PromiseLike<T>
   })
   ```
 
+::: tip NOTE
+You can set [chaiConfig](/config/#chaiconfig-truncatethreshold) configuration value to truncate the error assertion message.
+:::
 ## toMatchObject
 
 - **Type:** `(received: object | array) => Awaitable<void>`

--- a/docs/api/index.md
+++ b/docs/api/index.md
@@ -284,8 +284,8 @@ You cannot use this syntax, when using Vitest as [type checker](/guide/testing-t
 
   If you want to have access to `TestContext`, use `describe.each` with a single test.
 
-::: tip NOTE
-You can set [chaiConfig](/config/#chaiconfig-truncatethreshold) configuration value to truncate `test.each` title and error assertion message.
+::: tip
+Vitest processes `$values` with chai `format` method. If the value is too truncated, you can increase [chaiConfig.truncateThreshold](/config/#chaiconfig-truncatethreshold) in your config file.
 :::
 
 ::: warning

--- a/docs/api/index.md
+++ b/docs/api/index.md
@@ -284,6 +284,10 @@ You cannot use this syntax, when using Vitest as [type checker](/guide/testing-t
 
   If you want to have access to `TestContext`, use `describe.each` with a single test.
 
+::: tip NOTE
+You can set [chaiConfig](/config/#chaiconfig-truncatethreshold) configuration value to truncate `test.each` title and error assertion message.
+:::
+
 ::: warning
 You cannot use this syntax, when using Vitest as [type checker](/guide/testing-types).
 :::

